### PR TITLE
Java: Fuzzy Search on SAP HANA Cloud

### PR DESCRIPTION
**Pull request by MattSchur**

### CODE REVIEW

File: guides/providing-services.md [hunk 1]
```diff
@@ -424,6 +424,32 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
...
+
+```cds
+entity Books {
+  @Search.fuzzinessThreshold: 0.7
+  title : String;
+}
+```
+
+::: warning Only Java
+Fuzzy search on SAP HANA Cloud is currently only supported on the Java runtime and requires the [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode.
```
Comment (agoerler) [inline guides/providing-services.md:449]:
```suggestion
Fuzzy search on SAP HANA Cloud is currently only supported on the CAP Java runtime and requires the [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode.
```

File: guides/providing-services.md [hunk 2]
```diff
@@ -424,6 +424,32 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
 You can explicitly annotate calculated elements to make them searchable, even though they aren't searchable by default. The virtual elements won't be searchable even if they're explicitly annotated.
 :::
 
+#### Fuzzy Search on SAP HANA Cloud <Beta /> {#fuzzy-search}
+
+Fuzzy search is a fault-tolerant search feature of SAP HANA Cloud, which returns records even if the search term contains additional characters, is missing characters, or has typographical errors.
+
+If you run CAP Java in [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness (inclusive range 0-1, 1 is exact). The default fuzziness is 0.8.
```
Comment (agoerler) [inline guides/providing-services.md:431]:
```suggestion
If you run CAP Java in [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness (inclusive range 0.0-1.0, 1.0 is exact). The default fuzziness is 0.8.
```

or (with a closed interval notation):

```suggestion
If you run CAP Java in [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.
```

**Review by agoerler — APPROVED** [review]

File: guides/providing-services.md [hunk 3]
```diff
@@ -424,6 +424,37 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
 You can explicitly annotate calculated elements to make them searchable, even though they aren't searchable by default. The virtual elements won't be searchable even if they're explicitly annotated.
 :::
 
+#### Fuzzy Search on SAP HANA Cloud <Beta /> {#fuzzy-search}
+
+Fuzzy search is a fault-tolerant search feature of SAP HANA Cloud, which returns records even if the search term contains additional characters, is missing characters, or has typographical errors.
+
+If you run CAP Java in [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.
```
Comment (renejeglinsky) [inline guides/providing-services.md:431]:
```suggestion
If you run CAP Java in [`HEX ` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode) on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.
```

File: guides/providing-services.md [hunk 4]
```diff
@@ -424,6 +424,37 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
...
+}
+```
+
+::: tip Wildcards in search terms
+When using wildcards in search terms, an *exact pattern search* is performed.
+Supported wildcards are '*' matching zero or more characters and '?' matching a single character. You can escape wildcards using '\\'.
+:::
+
+::: warning Only Java
+Fuzzy search on SAP HANA Cloud is currently only supported on the CAP Java runtime and requires the [HEX](../java/cqn-services/persistence-services#sql-optimization-mode) mode.
```
Comment (renejeglinsky) [inline guides/providing-services.md:454]:
```suggestion
Fuzzy search on SAP HANA Cloud is currently only supported on the CAP Java runtime and requires the [`HEX ` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode).
```

File: guides/providing-services.md [hunk 5]
```diff
@@ -424,6 +424,37 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
...
+```cds
+entity Books {
+  @Search.fuzzinessThreshold: 0.7
+  title : String;
+}
+```
+
+::: tip Wildcards in search terms
+When using wildcards in search terms, an *exact pattern search* is performed.
+Supported wildcards are '*' matching zero or more characters and '?' matching a single character. You can escape wildcards using '\\'.
```
Comment (renejeglinsky) [inline guides/providing-services.md:450]: This information is important for users of the search, right? So the developer implementing fuzzy search would need to give this information to the user if the developer thinks that is helpful. Is that correct?

Comment (MattSchur) [reply guides/providing-services.md:450]: Yes, it's for all users of search, developers in event handlers and app users / clients performing search.

File: guides/providing-services.md [hunk 6]
```diff
@@ -424,6 +424,37 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
 You can explicitly annotate calculated elements to make them searchable, even though they aren't searchable by default. The virtual elements won't be searchable even if they're explicitly annotated.
 :::
 
+#### Fuzzy Search on SAP HANA Cloud <Beta /> {#fuzzy-search}
+
+Fuzzy search is a fault-tolerant search feature of SAP HANA Cloud, which returns records even if the search term contains additional characters, is missing characters, or has typographical errors.
+
+If you run CAP Java in [`HEX ` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode) on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.
```
Comment (MattSchur) [inline guides/providing-services.md:431]:
```suggestion
If you run CAP Java in [`HEX` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode) on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.
```

**Review by agoerler — APPROVED** [review]

File: guides/providing-services.md [hunk 7]
```diff
@@ -424,6 +424,37 @@ Searches all elements of type `String` excluding the element `isbn`, which leave
...
+}
+```
+
+::: tip Wildcards in search terms
+When using wildcards in search terms, an *exact pattern search* is performed.
+Supported wildcards are '*' matching zero or more characters and '?' matching a single character. You can escape wildcards using '\\'.
+:::
+
+::: warning Only Java
+Fuzzy search on SAP HANA Cloud is currently only supported on the CAP Java runtime and requires the [`HEX ` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode).
```
Comment (MattSchur) [inline guides/providing-services.md:454]:
```suggestion
Fuzzy search on SAP HANA Cloud is currently only supported on the CAP Java runtime and requires the [`HEX` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode).
```

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**